src/simsopt/mhd/virtual_casing.py

```diff
@@ -0,0 +1,355 @@
+        unit_normal[:, :, :] = self.unit_normal
+        unit_normal.description = 'Unit-length normal vector on the boundary surface'
+        unit_normal.units = 'Dimensionless'
+
+        B_total = f.createVariable('B_total', 'd', ('nphi', 'ntheta', 'xyz'))
+        B_total[:, :, :] = self.B_total
+        B_total.description = 'Total magnetic field vector on the surface, including currents both inside and outside of the surface'
+        B_total.units = 'Tesla'
+
+        B_internal = f.createVariable('B_internal', 'd', ('nphi', 'ntheta', 'xyz'))
+        B_internal[:, :, :] = self.B_internal
+        B_internal.description = 'Contribution to the magnetic field vector on the surface due only to currents inside the surface'
+        B_internal.units = 'Tesla'
+
+        B_internal_normal = f.createVariable('B_internal_normal', 'd', ('nphi', 'ntheta'))
+        B_internal_normal[:, :] = self.B_internal_normal
+        B_internal_normal.description = 'Component of B_internal normal to the surface'
+        B_internal_normal.units = 'Tesla'
+
+        f.close()
+
+    @classmethod
+    def load(cls, filename):
+        """
+        Load in the results of a previous virtual casing calculation,
+        previously saved in NetCDF format.
+
+        Args:
+            filename: Name of the file to load.
+        """
+        vc = cls()
+        f = netcdf_file(filename, mmap=False)
+        for key, val in f.variables.items():
+            val2 = val[()]  # Convert to numpy array
+            vc.__setattr__(key, val2)
+        return vc
+
+    def resample(self, ntheta, nphi):
+        """
+        Return a new ``VirtualCasing`` object in which all of the data
+        have been resampled in the poloidal and toroidal angles.
+        This is useful if you wish to use a different surface resolution
+        for the stage-2 coil optimization compared to the resolution
+        used for the virtual casing calculation.
+
+        Args:
+            ntheta: New number of grid points in the poloidal angle.
+            nphi: New number of grid points in the toroidal angle (including all field periods).
+
+        Returns:
+            A new ``VirtualCasing`` object.
+        """
+        newvc = VirtualCasing()
+        newvc.ntheta = ntheta
+        newvc.nphi = nphi
+        newvc.theta = np.linspace(0, 1, ntheta, endpoint=False)
+        newvc.phi = np.linspace(0, 1, nphi, endpoint=False)
+        newvc.B_internal_normal = resample_2D(self.B_internal_normal, nphi, ntheta)
+        # Vector fields on the surface:
+        variables = ['gamma', 'unit_normal', 'B_total', 'B_internal']
+        for variable in variables:
+            oldvar = eval('self.' + variable)
+            newvar = np.zeros((nphi, ntheta, 3))
+            for j in range(3):
+                newvar[:, :, j] = resample_2D(oldvar[:, :, j], nphi, ntheta)
+            newvc.__setattr__(variable, newvar)
+        return newvc
+
+    def plot(self, show=True):
```

Comment:
Saving the fig is not enabled. Also, the approach used in the plot method is archaic. Instead, OO-approach should be used to make it more convenient for end-users.

Please refer to https://matplotlib.org/stable/tutorials/introductory/lifecycle.html

Reply:
My suggestion is to make plot return fig, axes objects so users can further customize the plots. Then they can decide whether to save or show them.

Reply:
This has been addressed.